adapters/httpapi_curl.c

```diff
@@ -113,20 +113,87 @@ HTTP_HANDLE HTTPAPI_CreateConnection(const char* hostName)
         {
             size_t hostURL_size = strlen("https://") + strlen(hostName) + 1;
             httpHandleData->hostURL = malloc(hostURL_size);
+
             if (httpHandleData->hostURL == NULL)
             {
                 LogError("unable to malloc");
                 free(httpHandleData);
                 httpHandleData = NULL;
             }
+            else if ((strcpy_s(httpHandleData->hostURL, hostURL_size, "https://") != 0) ||
+                    (strcat_s(httpHandleData->hostURL, hostURL_size, hostName) != 0))
+            {
+                LogError("unable to set hostURL");
+                free(httpHandleData->hostURL);
+                free(httpHandleData);
+                httpHandleData = NULL;
+            }
+            else if ((httpHandleData->curl = curl_easy_init()) == NULL)
+            {
+                LogError("unable to init cURL structure");
+                free(httpHandleData->hostURL);
+                free(httpHandleData);
+                httpHandleData = NULL;
+            }
             else
             {
-                if ((strcpy_s(httpHandleData->hostURL, hostURL_size, "https://") == 0) &&
-                    (strcat_s(httpHandleData->hostURL, hostURL_size, hostName) == 0))
+                // Check correct TLS library has been configured for cURL.
+                const struct curl_tlssessioninfo *info = NULL;
```

Comment:
nit for this code base code style
```suggestion
                const struct curl_tlssessioninfo* info = NULL;
```